File: src/block_type.py
from enum import Enum

# Block Types Enum
class BlockType(Enum):
    EMPTY = 0
    GRASS = 1
    DIRT = 2
    STONE = 3
    WOOD = 4 
    LEAVES = 5

# Block Colors for Hotbar (RGB)
BLOCK_COLORS = {
    BlockType.GRASS: (0, 150, 0),
    BlockType.DIRT: (139, 69, 19),
    BlockType.STONE: (128, 128, 128),
    BlockType.WOOD: (160, 82, 45),
    BlockType.LEAVES: (0, 100, 0)
}

File: BlockType.js
// BlockType.js
export const BlockType = {
    // Order matches assumed vertical atlas: GRASS, DIRT, STONE, WOOD, LEAVES
    // atlasIndex: 0 for top texture (GRASS), 4 for bottom texture (LEAVES)
    // vOffset = (totalTextures - 1 - atlasIndex) * textureHeightRatio
    // textureHeightRatio = 1 / totalTextures
    EMPTY:  { id: 0, name: 'empty', atlasIndex: -1 }, // No texture
    GRASS:  { id: 1, name: 'grass', atlasIndex: 0 },
    DIRT:   { id: 2, name: 'dirt',  atlasIndex: 1 },
    STONE:  { id: 3, name: 'stone', atlasIndex: 2 },
    WOOD:   { id: 4, name: 'wood',  atlasIndex: 3 },
    LEAVES: { id: 5, name: 'leaves',atlasIndex: 4 }
};

export const ATLAS_NUM_TEXTURES = 5; // Total number of textures in the vertical strip

export function getBlockTypeById(id) {
    for (const key in BlockType) {
        if (BlockType[key].id === id) {
            return BlockType[key];
        }
    }
    return BlockType.EMPTY;
}
